src/services/upower/device.cpp

```diff
@@ -77,44 +76,48 @@ UPowerDevice::UPowerDevice(const QString& path, QObject* parent): QObject(parent
 	QObject::connect(&this->pTimeToFull, &AbstractDBusProperty::changed, this, &UPowerDevice::timeToFullChanged);
 	QObject::connect(&this->pPercentage, &AbstractDBusProperty::changed, this, &UPowerDevice::percentageChanged);
 	QObject::connect(&this->pIsPresent, &AbstractDBusProperty::changed, this, &UPowerDevice::isPresentChanged);
 	QObject::connect(&this->pState, &AbstractDBusProperty::changed, this, &UPowerDevice::stateChanged);
-	QObject::connect(&this->pCapacity, &AbstractDBusProperty::changed, this, &UPowerDevice::capacityChanged);
+	QObject::connect(&this->pHealthPercentage, &AbstractDBusProperty::changed, this, &UPowerDevice::healthPercentageChanged);
 	QObject::connect(&this->pIconName, &AbstractDBusProperty::changed, this, &UPowerDevice::iconNameChanged);
 
-	QObject::connect(&this->deviceProperties, &DBusPropertyGroup::getAllFinished, this, &UPowerDevice::onGetAllFinished);
+	QObject::connect(&this->deviceProperties, &DBusPropertyGroup::getAllFinished, this, &UPowerDevice::ready);
 	// clang-format on
 
 	this->deviceProperties.setInterface(this->device);
 	this->deviceProperties.updateAllViaGetAll();
 }
 
+// clang-format off
 bool UPowerDevice::isValid() const { return this->device->isValid(); }
 QString UPowerDevice::address() const { return this->device->service(); }
 QString UPowerDevice::path() const { return this->device->path(); }
 
 UPowerDeviceType::Enum UPowerDevice::type() const {
 	return static_cast<UPowerDeviceType::Enum>(this->pType.get());
 }
+
 bool UPowerDevice::powerSupply() const { return this->pPowerSupply.get(); }
 qreal UPowerDevice::energy() const { return this->pEnergy.get(); }
-qreal UPowerDevice::energyFull() const { return this->pEnergyFull.get(); }
-qreal UPowerDevice::energyRate() const { return this->pEnergyRate.get(); }
+qreal UPowerDevice::energyCapacity() const { return this->pEnergyCapacity.get(); }
+qreal UPowerDevice::changeRate() const { return this->pChangeRate.get(); }
 qlonglong UPowerDevice::timeToEmpty() const { return this->pTimeToEmpty.get(); }
 qlonglong UPowerDevice::timeToFull() const { return this->pTimeToFull.get(); }
 qreal UPowerDevice::percentage() const { return this->pPercentage.get(); }
 bool UPowerDevice::isPresent() const { return this->pIsPresent.get(); }
+
 UPowerDeviceState::Enum UPowerDevice::state() const {
 	return static_cast<UPowerDeviceState::Enum>(this->pState.get());
 }
-qreal UPowerDevice::capacity() const { return this->pCapacity.get(); }
-QString UPowerDevice::iconName() const { return this->pIconName.get(); }
 
-void UPowerDevice::onGetAllFinished() { emit this->ready(); }
+qreal UPowerDevice::healthPercentage() const { return this->pHealthPercentage.get(); }
+bool UPowerDevice::healthSupported() const { return this->path() != "/org/freedesktop/UPower/devices/DisplayDevice"; }
+QString UPowerDevice::iconName() const { return this->pIconName.get(); }
+// clang-format on
```

Comment:
nit: just let the formatter do what it wants here while keeping the line spacing correct